### PR TITLE
Adds documentation for the nodes in the data/dictionary namespace

### PR DESCRIPTION
**Pull request by amatiszcr**

### CODE REVIEW

**Review by danstocker — APPROVED** [review]

File: data/dictionary/item deleter (mutable)/85538251-c3cb-4dd8-abd5-920afaac7676.json [hunk 1]
```diff
@@ -1,21 +1,21 @@
 {
-  "description": "Deletes an item from the dictionary by it's key.\nMutates input.",
+  "description": "Deletes an item from the dictionary by it's key.\nMutates input.\n\nIf the item is not found, the original dictionary is forwarded.\n\nExample A:\n1. { \"first\": 1, \"third\": 3, \"fifth\": 5 } @0 received via `dict`\n2. \"first\"@0 received via `key`\n3. `dict` sends { \"third\": 3, \"fifth\": 5 }@0\n\nExample B:\n1. { \"first\": 1, \"third\": 3, \"fifth\": 5 } @0 received via `dict`\n2. \"second\"@0 received via `key`\n3. `dict` sends{ \"first\": 1, \"third\": 3, \"fifth\": 5 }@0\n\nMore:\nhttps://github.com/Cranq-io/cranq-tutorials/tree/main/reference/2_constructing_data/2_1_setters_deleters",
```
Comment (lvass74) [inline data/dictionary/item deleter (mutable)/85538251-c3cb-4dd8-abd5-920afaac7676.json:2]: Typo: "it"s key" should be "its key"

File: data/dictionary/item deleter/73a9fd6e-d594-4efc-a0ae-acf4ab89b577.json [hunk 2]
```diff
@@ -1,21 +1,24 @@
 {
   "name": "data/dictionary/Item deleter",
-  "description": "Deletes an item from the dictionary by it's key.",
+  "description": "Deletes an item from a dictionary by it's key. \nIf the item is not found, the original dictionary is forwarded.\n\nExample A:\n1. { \"first\": 1, \"third\": 3, \"fifth\": 5 } @0 received via `dict`\n2. \"first\"@0 received via `key`\n3. `dict` sends { \"third\": 3, \"fifth\": 5 }@0\n\nExample B:\n1. { \"first\": 1, \"third\": 3, \"fifth\": 5 } @0 received via `dict`\n2. \"second\"@0 received via `key`\n3. `dict` sends{ \"first\": 1, \"third\": 3, \"fifth\": 5 }@0\n\nMore:\nhttps://github.com/Cranq-io/cranq-tutorials/tree/main/reference/2_constructing_data/2_1_setters_deleters",
```
Comment (lvass74) [inline data/dictionary/item deleter/73a9fd6e-d594-4efc-a0ae-acf4ab89b577.json:3]: "it's key" should be "its key"

File: data/dictionary/item getter/857aacdb-326e-4b2e-a261-5197b667e14f.json [hunk 3]
```diff
@@ -1,24 +1,28 @@
 {
-  "description": "Retrieves an item's value from the dictionary.",
+  "description": "Retrieves an item's value by it's key from a dictionary.\nIf the item is not found, the inputs are sent via `not found`.\n\nExample A:\n1. { \"first\": 1, \"third\": 3, \"fifth\": 5 } @0 received via `dict`\n2. \"first\"@0 received via `key`\n3. `value` sends 1@0\n\nExample B:\n1. { \"first\": 1, \"third\": 3, \"fifth\": 5 } @0 received via `dict`\n2. \"second\"@0 received via `key`\n3. `not found` sends { \"dict\":  { \"first\": 1, \"third\": 3, \"fifth\": 5 }, \"key\": \"second\" }@0\n\nMore:\nhttps://github.com/Cranq-io/cranq-tutorials/tree/main/reference/3_querying_data/3_1_getters",
```
Comment (lvass74) [inline data/dictionary/item getter/857aacdb-326e-4b2e-a261-5197b667e14f.json:2]: "its key"

File: data/dictionary/item setter/3966e767-b7a7-4221-961d-338a13d18c57.json [hunk 4]
```diff
@@ -1,24 +1,28 @@
 {
-  "description": "Sets an item in the dictionary.",
+  "description": "Sets an item's value by it's key in a dictionary.\nIf the item is not found, it will be created.\n\nExample A:\n1. { \"first\": 1, \"third\": 3, \"fifth\": 5 } @0 received via `dict`\n2. \"first\"@0 received via `key`\n3. -1@0 received via `value`\n4. `dict` sends { \"first\": -1, \"third\": 3, \"fifth\": 5 }@0\n\nExample B:\n1. { \"first\": 1, \"third\": 3, \"fifth\": 5 } @0 received via `dict`\n2. \"second\"@0 received via `key`\n3. 2@0 received via `value`\n4. `dict` sends { \"first\": 1, \"second\":2, \"third\": 3, \"fifth\": 5 }@0\n\nMore:\nhttps://github.com/Cranq-io/cranq-tutorials/tree/main/reference/2_constructing_data/2_1_setters_deleters",
```
Comment (lvass74) [inline data/dictionary/item setter/3966e767-b7a7-4221-961d-338a13d18c57.json:2]: "its key"

File: data/dictionary/items getter/83ac4d5d-9ac9-48d0-ae17-a5a77b18cfb6.json [hunk 5]
```diff
@@ -1,20 +1,23 @@
...
-        "name": "keys"
+        "name": "keys",
+        "description": "Receives the keys corresponding to the values to get.\n\nExample:\n[\"first\", \"third\"]"
       }
     },
     "outputs": {
       "6dd99f55-3d65-42ca-b301-c2c6367c334c": {
-        "type": "typeof(dict)[string][]",
-        "name": "values"
+        "type": "typeof`dict`[string][]",
```
Comment (lvass74) [inline data/dictionary/items getter/83ac4d5d-9ac9-48d0-ae17-a5a77b18cfb6.json:18]: As the dictionary"s values are of type any, I guess the `values` output port's type should be `any[]`. Not sure if this expression actually means the same.

File: data/dictionary/json parser/6470bc83-fb15-4801-942a-6f6dbc46a908.json [hunk 6]
```diff
@@ -1,20 +1,23 @@
...
     },
     "outputs": {
       "6cdc0c38-5098-4ffd-a5d8-ee18d077dda3": {
         "type": "{string:any}",
-        "name": "parsed"
+        "name": "parsed",
+        "description": "Sends the parsed dictionary.\n\nExample:\n{ \"first\": 1, \"third\": 3, \"fifth\": 5 }"
       },
       "426110b5-257a-45c2-976c-f42195dafde8": {
         "type": "string",
```
Comment (lvass74) [inline data/dictionary/json parser/6470bc83-fb15-4801-942a-6f6dbc46a908.json:18]: Tip: can be typeof `json`

File: data/dictionary/left joiner/20ab24f4-9017-4cf1-9b44-45211ec3d074.json [hunk 7]
```diff
@@ -1,20 +1,23 @@
 {
-  "description": "Joins dictionary 'b' to dictionary 'a', by matching the values of 'a' to the keys of 'b'.",
+  "description": "Joins dictionary 'b' to dictionary `a`, by matching the values of `a` to the keys of `b`.\n\nItems from `b` with no matching value in `a` are excluded from the result.\n\nItems from `a` with no matching key in `b` are excluded from the result.\n\nExample:\n1. {\"first\":\"first_id\", \"third\":\"third_id\", \"fifth\":\"fifth_id\"}@0 received via `a`\n2. {\"first_id\":1, \"second_id\":2, \"third_id\":3}@0 received via `b`\n3. `joined` sends {\"first\": 1,\"third\": 3}@0",
   "interface": {
     "inputs": {
       "dfab5f2f-9107-481d-b7b1-19a7631381fb": {
         "type": "{string:string}",
-        "name": "a"
+        "name": "a",
+        "description": "Receives the dictionary containing the keys to join to, and the values to join by.\n\nExample:\n{\"first\":\"first_id\", \"third\":\"third_id\", \"fifth\":\"fifth_id\"}"
```
Comment (lvass74) [inline data/dictionary/left joiner/20ab24f4-9017-4cf1-9b44-45211ec3d074.json:8]: I think "," is not necessary after "the keys to join to"

File: data/dictionary/left joiner/20ab24f4-9017-4cf1-9b44-45211ec3d074.json [hunk 8]
```diff
@@ -1,20 +1,23 @@
...
       },
       "be713809-1c65-4abb-9525-87ee9f75e9be": {
         "type": "{string:any}",
-        "name": "b"
+        "name": "b",
+        "description": "Receives the dictionary containing the values to join to, and the keys to join by.\n\nExample:\n{\"first_id\":1, \"second_id\":2, \"third_id\":3}"
```
Comment (lvass74) [inline data/dictionary/left joiner/20ab24f4-9017-4cf1-9b44-45211ec3d074.json:13]: I think the correct description would be:
"Receives the dictionary containing the keys to join by and the values to join."
... unless I misunderstood :)

File: data/dictionary/left joiner/20ab24f4-9017-4cf1-9b44-45211ec3d074.json [hunk 9]
```diff
@@ -1,20 +1,23 @@
...
       }
     },
     "outputs": {
       "81c6113a-17f2-4f3b-840b-51a7f1026d3d": {
-        "type": "{string:typeof(b)[string]}",
-        "name": "joined"
+        "type": "{string:typeof`b`[string]}",
```
Comment (lvass74) [inline data/dictionary/left joiner/20ab24f4-9017-4cf1-9b44-45211ec3d074.json:18]: Does this means the same as {string:any} ?

File: data/dictionary/values getter/3419d082-1bea-4e6b-8dfb-47eabfe792b0.json [hunk 10]
```diff
@@ -1,16 +1,18 @@
 {
-  "description": "Gets values from the dictionary as an array.",
+  "description": "Gets all values from the dictionary as an array.\n\n\nExample A:\n1. { \"first\": 1, \"third\": 3, \"fifth\": 5 } @0 received via `dict`\n2. `values` sends  [ 1,3,5 ]@0\n\nMore:\nhttps://github.com/Cranq-io/cranq-tutorials/tree/main/reference/3_querying_data/3_1_getters",
   "interface": {
     "inputs": {
       "8a82bc3e-2765-4d8b-b372-fbb77a4f4383": {
         "type": "{string:any}",
-        "name": "dict"
+        "name": "dict",
+        "description": "Receives to dictionary to extract the values from.\n\nExample:\n{ \"first\": 1, \"third\": 3, \"fifth\": 5 } "
```
Comment (lvass74) [inline data/dictionary/values getter/3419d082-1bea-4e6b-8dfb-47eabfe792b0.json:8]: Receives **the** dictionary

**Review by lvass74 — CHANGES_REQUESTED** [review]
Submitted a couple of comments regarding typos or text corrections.

File: data/dictionary/items getter/83ac4d5d-9ac9-48d0-ae17-a5a77b18cfb6.json (hunk 5 above)
Comment (lvass74) [reply data/dictionary/items getter/83ac4d5d-9ac9-48d0-ae17-a5a77b18cfb6.json:18]: That's fine, we agreed that output ports should refer to input port's type whenever possible to keep types statically analyzable.
So don't change this.

File: data/dictionary/left joiner/20ab24f4-9017-4cf1-9b44-45211ec3d074.json (hunk 9 above)
Comment (lvass74) [reply data/dictionary/left joiner/20ab24f4-9017-4cf1-9b44-45211ec3d074.json:18]: It's fine, don't change it.

**Comment by amatiszcr** [comment]
> Submitted a couple of comments regarding typos or text corrections.

fixed

**Pull request merged**